common/hash_test.go

```diff
@@ -17,14 +17,49 @@
 package common
 
 import (
+	"bytes"
 	"testing"
 )
 
+func TestBlake2b218(t *testing.T) {
+	in := []byte{0x1}
+	h, err := Blake2b128(in)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expected, err := HexToBytes("0x01cae66941d9efbd404e4d88758ea676")
```

Comment:
Please stop putting unexplained magical values into tests. Without any reference to their source they cannot be verified (or changed where applicable). Please add comments indicating the input value/implementation used for these

Reply:
You could possibly be using the go implementations instead of providing precomputed string values

Reply:
what do you mean using the go implementations?

Reply:
I have created a test repo that calculated blake2b hashes for 128 and 256 using blake2-rfc (the crate substrate uses) https://github.com/noot/blake2b_test

https://github.com/paritytech/substrate/blob/master/core/primitives/src/hashing.rs

Reply:
the test cases are from the rust library blake2-rfp that substrate uses: https://github.com/noot/blake2b_test/blob/master/src/main.rs